scipy/stats/tests/test_axis_nan_policy.py

```diff
@@ -866,15 +868,16 @@ def small_sample_generator(n_dims):
                     sup.filter(RuntimeWarning, "Mean of empty slice.")
                     sup.filter(RuntimeWarning, "invalid value encountered")
                     expected = np.mean(concat, axis=axis) * np.nan
+                    mask = np.isnan(expected)
+                    expected = [np.asarray(expected.copy()) for i in range(n_outputs)]
```

Comment:
The purpose of this part of `assert_equal(res[i], expected[i])` is to check that empty slices produce the intended statistic and p-value. Previously, the only function that didn't produce NaN with empty input was `entropy`, and that has only one output. (For example, see below that `expected[mask] = empty_val`, but `empty_val` is the output of `hypotest`, so if it were a tuple, we'd have issues.) So this needs to be generalized for multiple outputs.

This line assumes that all expected values are NaN; we adjust that for special case functions below.